karapace/protobuf/dependency.py

```diff
@@ -35,13 +35,32 @@ def add_used_type(self, parent: str, element_type: str) -> None:
     def add_import(self, import_name: str) -> None:
         self.import_path.append(import_name)
 
+    def is_type_declared(
+        self,
+        used_type: str,
+        declared_index: Set[str],
+        delimiter: int,
+        father_child_type: Optional[str],
+        used_type_with_scope: str,
+    ) -> bool:
+        return (
+            used_type in declared_index
+            or (delimiter != -1 and used_type_with_scope in declared_index)
+            or (father_child_type is not None and father_child_type in declared_index)
+            or "." + used_type in declared_index
+        )
+
     def verify(self) -> DependencyVerifierResult:
         declared_index = set(self.declared_types)
         for used_type in self.used_types:
             delimiter = used_type.rfind(";")
+            father_child_type = None
             used_type_with_scope = ""
```

Comment:
This `used_type_with_scope` could work similar as the `father_child_type`, set to `None` and then there is no need to pass `delimiter` to `is_type_declared`.